File: day1_rocket_equation/python/day1_rocket_equation.py
#!/bin/python

import sys
import io

file_path = sys.argv[1]
masses = []
with io.open(file_path, "r") as f:
    while True:
        line = f.readline()
        if line:
            masses.append(int(line.strip()))
        else:
            break

total_fuel = sum([(m // 3) - 2 for m in masses])

print("{}".format(total_fuel))

File: day1_rocket_equation/rust/day1_rocket_equation.rs
use std::env;
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let args: Vec<String> = env::args().collect();
    let file_path: &String = &args[1];

    let f = File::open(file_path).expect("Unable to open file");
    let f = BufReader::new(f);

    let mut masses = Vec::new();
    for line in f.lines() {
        let mass: i32 = line.expect("Unable to read line").trim().parse::<i32>().expect("Can't parse");
        masses.push(mass);
    }
    let total_fuel = masses.iter().fold(0, |sum, x| sum + (x / 3) - 2);
    
    println!("{}", total_fuel);
}
